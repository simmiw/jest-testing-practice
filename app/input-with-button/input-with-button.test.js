import InputWithButton from "./page";
import { fireEvent, screen, render } from "@testing-library/react";

describe("inputWithButton", () => {
  it("should render form", async () => {
    const sut = render(<InputWithButton />);
    expect(sut.container).toMatchSnapshot();
  });

  it("should render the input, Button and list", () => {
    render(<InputWithButton />);
    fireEvent.change(screen.getByPlaceholderText("Enter any text"), {
      target: {
        value: "Hello",
      },
    });
    fireEvent.click(screen.getByText(/Submit/));
    expect(screen.getByText("Hello")).toBeVisible();
  });
});
