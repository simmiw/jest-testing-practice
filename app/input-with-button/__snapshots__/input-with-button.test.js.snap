// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputWithButton should render form 1`] = `
<div>
  <form>
    <input
      placeholder="Enter any text"
      type="text"
      value=""
    />
    <button>
      Click to Submit
    </button>
  </form>
  <ul />
</div>
`;
